salt/modules/file.py

```diff
@@ -1884,13 +1886,13 @@ def line(path, content=None, match=None, mode=None, location=None,
                 _assert_occurrence(body, after, 'after')
                 out = []
                 lines = body.split(os.linesep)
-                for idx in range(len(lines)):
-                    _line = lines[idx]
-                    if _line.find(before) > -1 and idx <= len(lines) and lines[idx - 1].find(after) > -1:
-                        out.append(_get_line_indent(_line, content, indent))
-                        out.append(_line)
-                    else:
-                        out.append(_line)
+                in_range = False
+                for idx, line in enumerate(lines):
```

Comment:
Your modifications no longer use the `idx` variable, so there is no need to do an `enumerate` here, and you can simply iterate over the list (e.g. `for line in lines`).